pennylane/transforms/unbroadcast_expand.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the tape expansion function for unbroadcasting a
+broadcasted tape into multiple tapes."""
+import pennylane as qml
+from pennylane.transforms.batch_transform import batch_transform
+
+
+@batch_transform
+def unbroadcast_expand(tape):
+    r"""Expand a broadcasted tape into multiple unbroadcasted tapes
+    and a function that stacks and squeezes the results.
```

Comment:
Yeah I think this is where I get confused. Broadcasting to me is a verb; a tape can have ndim parameters, but it can't be described as 'broadcasted' before anything happens. Broadcasting only happens when you expand 🤔

Reply:
In a way, the broadcasting _is_ the expansion?

Reply:
This is a tough one regarding name space :thinking: 
>but it can't be described as 'broadcasted' before anything happens
I think my understanding differs at this point, leading to the confusion. `qml.RX([1, 2], wires=0)` in my understanding broadcasts the operator across the additional parameter axis, and I thought broadcasting the operator makes the operator broadcasted. Looking at NumPy for a comparison again, broadcasting would mean to me to do 
`np.ones((3, 4)) * np.ones(4)` whereas the result of this expansion would correspond to duplicating the operation, yielding `np.array([row * np.ones(4) for row in np.ones((3, 4))])`.

I'm not sure how to be clearer here? :thinking: Happy for any suggestions you might have :)

Reply:
I think then it should be fine, together with removing the inconvenient "unbroadcast" naming, right? :)

Reply:
yep!